TombEngine/Game/missile.cpp

```diff
@@ -188,18 +189,19 @@ short ShardGun(int x, int y, int z, short velocity, short yRot, short roomNumber
 
 short BombGun(int x, int y, int z, short velocity, short yRot, short roomNumber)
 {
-	int fxNumber = CreateNewEffect(roomNumber);
+	int fxNumber = CreateNewEffect(roomNumber, ID_PROJ_BOMB, Pose(x, y, z));
 	if (fxNumber != NO_ITEM)
 	{
-		auto& fx = EffectList[fxNumber];
-
-		fx.pos.Position = Vector3i(x, y, z);
-		fx.pos.Orientation = EulerAngles(0, yRot, 0);
-		fx.roomNumber = roomNumber;
-		fx.speed = velocity;
-		fx.frameNumber = 0;
-		fx.objectNumber = ID_PROJ_BOMB;
-		fx.color = Vector4::One;
+		auto& fx = g_Level.Items[fxNumber];
+
+		fx.Pose.Position = Vector3i(x, y, z);
+		fx.Pose.Orientation = EulerAngles(0, yRot, 0);
+		fx.RoomNumber = roomNumber;
+		fx.Animation.Velocity.z = velocity;
+		fx.Animation.FrameNumber = 0;
+		fx.ObjectNumber = ID_PROJ_BOMB;
```

Comment:
Do we need to re-assign object number if it was already done in `CreateNewEffect` above?